components/contexts/NavigationContext.tsx

```diff
@@ -0,0 +1,60 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import useWindow from "~/config/hooks/useWindow";
+import { breakpoints } from "~/config/themes/screen";
+
+type NavContextType = {
+  isRefManagerSidebarOpen: boolean;
+  setIsRefManagerSidebarOpen: (isOpen: boolean) => void;
+  isRefManagerDisplayedAsDrawer: boolean;
+  setIsRefManagerDisplayedAsDrawer: (isDrawer: boolean) => void;
+};
+
+const NavigationContext = createContext<NavContextType>({
+  isRefManagerSidebarOpen: false,
+  setIsRefManagerSidebarOpen: (isOpen) => {
+    throw new Error("setIsRefManagerSidebarOpen not implemented");
+  },
+  isRefManagerDisplayedAsDrawer: false,
+  setIsRefManagerDisplayedAsDrawer: (isOpen) => {
+    throw new Error("setIsRefManagerDisplayedAsDrawer not implemented");
+  },
+});
+
+export const navContext = () => useContext(NavigationContext);
+
+export const NavigationContextProvider = ({ children }) => {
+  const [isRefManagerSidebarOpen, setIsRefManagerSidebarOpen] = useState(false);
+  const [isRefManagerDisplayedAsDrawer, setIsRefManagerDisplayedAsDrawer] =
+    useState(false);
+
+  const { width: winWidth, height: winHeight } = useWindow();
+
+  useEffect(() => {
+    const displayAsDrawer = winWidth && winWidth < breakpoints.medium.int;
+
+    if (displayAsDrawer && !isRefManagerDisplayedAsDrawer) {
+      setIsRefManagerDisplayedAsDrawer(true);
```

Comment:
Typically I like handling this in CSS instead of JS if we can. 

Having CSS depend on JS can lead to bugs / flashing issues.

Is it necessary to handle these widths here like this?